toolchain/parse/parse.cpp

```diff
@@ -44,6 +45,7 @@ auto Parse(Lex::TokenizedBuffer& tokens, DiagnosticConsumer& consumer,
   }
 
   context.AddLeafNode(NodeKind::FileEnd, *context.position());
+  tree.set_has_errors(err_tracker.seen_error());
```

Comment:
Checking my understanding here:
- On the tree as a whole, `has_errors` indicates whether the input was syntactically invalid.
- On individual parse nodes, `has_error` indicates whether the node satisfies its kind's invariants, and may be false even if it corresponds to a syntactically invalid input.

If that's right, we should probably rename `has_error` on nodes to something like `well_formed`, but that's out of scope for this PR.

Reply:
Maybe I'd do a separate pass at refactoring both, but I actually think a node's `has_error` is the more correct one. The node has an error in its immediate structure (non-transitive from children), typically because parse encountered a problem it couldn't recover from.

OTOH, `has_errors` can include any diagnostic that should prevent lowering. That goes beyond invalid syntax. It's probably more the one that should be renamed, but Lex and Check both use the same name, so similar scope issue.